nerfacc/cuda/csrc/grid.cu

```diff
@@ -237,6 +237,177 @@ __global__ void traverse_grids_kernel(
 }
 
 
+__global__ void traverse_grids_test_kernel(
+    const int32_t N_samples,
+    int32_t *num_steps,
+    // rays
+    int32_t n_rays,
+    float *rays_o,  // [n_rays, 3]
+    float *rays_d,  // [n_rays, 3]
+    // grids
+    int32_t n_grids,
+    int3 resolution,
+    bool *binaries, // [n_grids, resx, resy, resz]
+    float *aabbs,   // [n_grids, 6]
+    // sorted intersections
+    bool *hits,         // [n_rays, n_grids]
+    float *t_sorted,    // [n_rays, n_grids * 2]
+    int64_t *t_indices, // [n_rays, n_grids * 2]
+    bool *continuous_resume, // [n_rays]
+    // options
+    float *near_planes,
+    float *far_planes,
+    float step_size,
+    float cone_angle,
+    // outputs
+    float *t_starts,
+    float *t_ends,
+    bool *valid_mask,
+    int64_t *ray_indices)
+{
+    float eps = 1e-6f;
+
+    // parallelize over rays
+    for (int32_t tid = blockIdx.x * blockDim.x + threadIdx.x; tid < n_rays; tid += blockDim.x * gridDim.x)
+    {
+
+        num_steps += tid;
+        continuous_resume += tid;
+        t_starts += tid * N_samples;
+        t_ends += tid * N_samples;
+        valid_mask += tid * N_samples;
+        ray_indices += tid * N_samples;
```

Comment:
Can't use `+=` anymore because of the forloop:

    for (int32_t tid = blockIdx.x * blockDim.x + threadIdx.x; tid < n_rays; tid += blockDim.x * gridDim.x)

It would be wrong if `+=` is executed twice within the forloop.

use something like `valid_mask[tid * Nsamples]` to read/write the data